choreolib/src/main/java/com/choreo/lib/ChoreoTrajectoryState.java

```diff
@@ -91,14 +104,24 @@ public ChoreoTrajectoryState interpolate(ChoreoTrajectoryState endValue, double
     double scale = (t - this.timestamp) / (endValue.timestamp - this.timestamp);
     var interp_pose = getPose().interpolate(endValue.getPose(), scale);
 
+    // NOTE: Could maybe do this with streams? This seems more efficient for now.
+    double[] interp_fx = new double[4];
+    double[] interp_fy = new double[4];
+    for (int i = 0; i < 4; ++i) {
+      interp_fx[i] = MathUtil.interpolate(this.moduleForcesX[i], endValue.moduleForcesX[i], scale);
+      interp_fy[i] = MathUtil.interpolate(this.moduleForcesY[i], endValue.moduleForcesY[i], scale);
```

Comment:
If streams or some form of map could avoid allocating two new Arrays here that would be nice

Reply:
Streams don't necessarily avoid allocation. An array of primitives should end up being stack-allocated.

Reply:
This was my thinking. The overhead of streams isn't worth it here IMO, especially given how simple the logic is.